src/simulation/LoadGenerator.cpp

```diff
@@ -866,11 +1025,224 @@ LoadGenerator::manageOfferTransaction(
+    createResources.instructions = 200'000;
+    createResources.readBytes = 5000;
+    createResources.writeBytes = 5000;
+
+    SCVal scContractSourceRefKey(SCValType::SCV_LEDGER_KEY_CONTRACT_INSTANCE);
+    auto salt = sha256(std::to_string(accountId));
+    auto [createOp, contractID] =
+        createSorobanCreateOp(mApp, createResources, *mCodeKey, *account,
+                              scContractSourceRefKey, salt);
+
+    auto const& instanceLk = createResources.footprint.readWrite.back();
+    ContractInstance instance;
+    instance.readOnlyKeys.emplace_back(*mCodeKey);
+    instance.readOnlyKeys.emplace_back(instanceLk);
+    instance.contractID = instanceLk.contractData().contract;
+    mContractInstances.emplace(accountId, instance);
+
+    auto resourceFee = sorobanResourceFee(mApp, createResources, 1000, 40);
+    resourceFee += 1'000'000;
+    auto tx = std::dynamic_pointer_cast<TransactionFrame>(
+        sorobanTransactionFrameFromOps(mApp.getNetworkID(), *account,
+                                       {createOp}, {}, createResources,
+                                       inclusionFee, resourceFee));
+
+    return std::make_pair(account, tx);
+}
+
+std::pair<LoadGenerator::TestAccountPtr, TransactionFramePtr>
+LoadGenerator::invokeSorobanLoadTransaction(uint32_t ledgerNum,
+                                            uint64_t accountId,
+                                            GeneratedLoadConfig const& cfg)
+{
+    auto account = findAccount(accountId, ledgerNum);
+    auto instanceIter = mContractInstances.find(accountId);
+    releaseAssert(instanceIter != mContractInstances.end());
+    auto const& instance = instanceIter->second;
+
+    auto const& networkCfg = mApp.getLedgerManager().getSorobanNetworkConfig();
+
+    // Approximate instruction measurements from loadgen contract. While the
+    // guest and host cycle counts are accurate, it very difficult to estimate
+    // the number of instructions that the write loops will consume
+    // without preflight, so some TXs will fail due to exceeding resource
+    // limitations. However these should fail at apply time, so will still
+    // generate siginificant load
+    uint64_t const baseInstructionCount = 3'000'000;
+    uint64_t const instructionsPerGuestCycle = 120;
+    uint64_t const instructionsPerHostCycle = 2355;
+
+    // Pick random number of cycles between bounds, respecting network limits
+    uint64_t maxInstructions =
+        networkCfg.mTxMaxInstructions - baseInstructionCount;
+
+    uint64_t guestCyclesMax = maxInstructions / instructionsPerGuestCycle;
+    guestCyclesMax = std::min(cfg.guestCyclesHigh, guestCyclesMax);
+    uint64_t guestCyclesMin = std::min(cfg.guestCyclesLow, guestCyclesMax);
+    uint64_t guestCycles =
+        rand_uniform<uint64_t>(guestCyclesMin, guestCyclesMax);
+
+    maxInstructions -= guestCycles * instructionsPerGuestCycle;
+    uint64_t hostCyclesMax = maxInstructions / instructionsPerHostCycle;
```

Comment:
It's a bit hard to argue about the distribution properties here (I can't say much about the instructions distribution while looking at this). I believe a simpler option would be to generate the 'target' number of instructions first, then split randomly between guest and host (i.e. generate guest cycles and dedicate the remainder to the host cycles). This way we'll be able to say that instructions are (roughly) uniformly distributed.